pilot/cmd/pilot-agent/options/security.go

```diff
@@ -52,6 +52,7 @@ func NewSecurityOptions(proxyConfig *meshconfig.ProxyConfig, stsPort int, tokenM
 		SecretRotationGracePeriodRatio: secretRotationGracePeriodRatioEnv,
 		STSPort:                        stsPort,
 		CertSigner:                     certSigner.Get(),
+		CARootPath:                     security.GetOSRootFilePath(),
```

Comment:
also it's confused to me that

following the code path, ca root path is only set here in the options, at the startup time, right? and then later on, this field is passed to GenerateSecret() in sdsservice.go without ever changing.

why we have to parameterize it in this case? would it work if we provide an `sync.Once` initialization and later on always use that value? My understanding is that in the startup, we can do init to detect what ca cert is available and keep using it from that point on.

Reply:
Getting the OS CA file path was moved to init but with that I don't think `sync.Once` was needed but I would be happy to add it if you find it necessary. This removes the added parameter in `GenerateSecret()`.